File: qc-domain/src/main/java/com/eager/questioncloud/domain/post/Post.java
package com.eager.questioncloud.domain.post;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
public class Post {
    private Long id;
    private Long questionId;
    private Long writerId;
    private PostContent postContent;
    private LocalDateTime createdAt;

    @Builder
    public Post(Long id, Long questionId, Long writerId, PostContent postContent, LocalDateTime createdAt) {
        this.id = id;
        this.questionId = questionId;
        this.writerId = writerId;
        this.postContent = postContent;
        this.createdAt = createdAt;
    }

    public static Post create(Long questionId, Long writerId, PostContent postContent) {
        return Post.builder()
            .questionId(questionId)
            .writerId(writerId)
            .postContent(postContent)
            .createdAt(LocalDateTime.now())
            .build();
    }

    public void updateQuestionBoardContent(PostContent postContent) {
        this.postContent = postContent;
    }
}

File: qc-domain/qc-post/src/main/kotlin/com/eager/questioncloud/post/domain/Post.kt
package com.eager.questioncloud.post.domain

import java.time.LocalDateTime

class Post(
    val id: Long = 0,
    val questionId: Long,
    val writerId: Long,
    var postContent: PostContent,
    val createdAt: LocalDateTime = LocalDateTime.now(),
) {
    fun updateQuestionBoardContent(postContent: PostContent) {
        this.postContent = postContent
    }

    companion object {
        fun create(questionId: Long, writerId: Long, postContent: PostContent): Post {
            return Post(questionId = questionId, writerId = writerId, postContent = postContent)
        }
    }
}
